static/app/views/automations/components/dataConditionNodes.tsx

```diff
@@ -0,0 +1,149 @@
+import {createContext, useContext} from 'react';
+import styled from '@emotion/styled';
+
+import NumberField from 'sentry/components/forms/fields/numberField';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import {t, tct} from 'sentry/locale';
+import {
+  type DataCondition,
+  DataConditionType,
+} from 'sentry/types/workflowEngine/dataConditions';
+
+interface DataConditionNodeProps {
+  condition: Omit<DataCondition, 'condition_group' | 'type' | 'id'>;
+  condition_id: string;
+  onUpdate: (condition: Record<string, any>) => void;
+}
+
+export const DataConditionNodeContext = createContext<DataConditionNodeProps | null>(
+  null
+);
+
+export function useDataConditionNodeContext(): DataConditionNodeProps {
+  const context = useContext(DataConditionNodeContext);
+  if (!context) {
+    throw new Error(
+      'useDataConditionNodeContext was called outside of DataConditionNode'
+    );
+  }
+  return context;
+}
+
+type Node = {
+  label: string;
+  configNode?: () => React.ReactNode;
```

Comment:
`configNode` seems extremely generic to the point of being confusing. Can you think of any better options? Maybe `<DataConditionField />` or `renderDataCondition()`?

Reply:
Changed this to `dataCondition` in https://github.com/getsentry/sentry/pull/89233/commits/3d1933f98380f8ca3367263cd30fe7586169a924 🫡  This doesn't need to be a function anymore since I created functions for each condition in their own individual files